File: Data Structures & Algorithms/max-water-container/submission-7.java
class Solution {
    public int maxArea(int[] heights) {
        int res = 0;
        for (int i = 0; i < heights.length; i++) {
            for (int j = i + 1; j < heights.length; j++) {
                int currArea = Math.min(heights[i], heights[j]) * (j - i);
                res = Math.max(res, currArea);
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-7.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = l = 0
        counts = {}

        for r in range(len(s)):
            if s[r] in counts:
                counts[s[r]] += 1
            else:
                counts[s[r]] = 1

            while (r - l + 1) - max(counts.values()) > k:
                counts[s[l]] -= 1
                l += 1
        
            res = max(res, r - l + 1)

        return res


